src/Range.tsx

```diff
@@ -55,6 +57,7 @@ class Range extends React.Component<IProps> {
     this.schdOnMouseMove = schd(this.onMouseMove);
     this.schdOnTouchMove = schd(this.onTouchMove);
     this.schdOnEnd = schd(this.onEnd);
+    this.values = props.values
```

Comment:
Is this really needed? `componentDidUpdate` can use `props.values` and the `checkValueAgainstBoundaries ` map could be directly passed into `setValues` (or `onChange`, not sure why a new API is needed)